### PR TITLE
feat: Billing API integration

### PR DESCRIPTION
**Pull request by JUNIOR140889**
This PR includes:

 * Purchase flow integration
 
Evidence:

https://github.com/JUNIOR140889/Black-Market-RN/assets/42948405/db147d4d-2d69-488d-bda1-6ab8bd15ddb6

### CODE REVIEW

**Review by fernandatoledo — APPROVED** [review]

File: src/api/purchase/use-purchase.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+type Order = {
+  order: OrderVariables;
+};
+
+export const purchaseOrder = async (order: Order): Promise<Response> => {
+  try {
+    const { data } = await client.post('/orders', order);
+    console.log(data);
```
Comment (ramirogavagnin) [inline src/api/purchase/use-purchase.tsx]: `console.log()`

File: src/api/purchase/use-purchase.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  shipping_address: ShippingAddres;
+};
+
+type Order = {
+  order: OrderVariables;
+};
+
+export const purchaseOrder = async (order: Order): Promise<Response> => {
+  try {
+    const { data } = await client.post('/orders', order);
```
Comment (ramirogavagnin) [inline src/api/purchase/use-purchase.tsx]: maybe you could have a constant with the endpoints, `const PURCHASE_ENDPOINT = { orders: '/orders' }` on `purchase/constants.ts`

File: src/api/purchase/use-purchase.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import type { UseMutationOptions } from '@tanstack/react-query';
```
Comment (ramirogavagnin) [inline src/api/purchase/use-purchase.tsx]: just a general thing, if you're not returning a `jsx` component just use `ts` extention intead of `tsx`

File: src/modules/payment/screens/payment-screen.tsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import React, { useRef } from 'react';
 import { Image as RNImage } from 'react-native';
 import { Checkbox } from 'react-native-paper';
 
+import { usePurchaseOrder } from '../../../api/purchase/use-purchase';
```
Comment (ramirogavagnin) [inline src/modules/payment/screens/payment-screen.tsx:7]: just a suggestion, maybe we could simplify the imports using path alias, leave you a post here -> https://javascript.plainenglish.io/react-native-path-aliasing-with-simplify-your-imports-and-enhance-your-codebase-9897efee96a8

**Review by ramirogavagnin — APPROVED** [review]

**Review by ramirogavagnin — APPROVED** [review]
nice job! I think you need to rebase with the target branch ;)

File: src/modules/payment/screens/payment-screen.tsx [hunk 5]
```diff
@@ -22,8 +22,28 @@ export const PaymentScreen: React.FC = () => {
...
+          city: data.city,
+          country: data.country,
+          line_1: data.adress1,
+          line_2: typeof data.adress2 === 'undefined' ? '-' : data.adress2,
+          postal_code: data.postalCode,
+          state: data.country,
+        },
+      },
+    });
+    navigation.navigate("Tab", {screen: "Home"});
```
Comment (juanchoperezj) [inline src/modules/payment/screens/payment-screen.tsx]: Consider moving this navigate method to the onSuccess callback in the `usePurchaseOrder`, with that you are sure that the user is going to the Home screen if the order was successfully placed.

**Review by juanchoperezj — APPROVED** [review]
Great work, just a small comment

File: src/api/purchase/use-purchase.tsx (hunk 1 above)
Comment (JUNIOR140889) [reply src/api/purchase/use-purchase.tsx]: Done

File: src/api/purchase/use-purchase.tsx (hunk 2 above)
Comment (JUNIOR140889) [reply src/api/purchase/use-purchase.tsx]: Done

File: src/api/purchase/use-purchase.tsx (hunk 3 above)
Comment (JUNIOR140889) [reply src/api/purchase/use-purchase.tsx]: Done

File: src/modules/payment/screens/payment-screen.tsx (hunk 4 above)
Comment (JUNIOR140889) [reply src/modules/payment/screens/payment-screen.tsx:7]: I will solve that in the next PR

File: src/modules/payment/screens/payment-screen.tsx (hunk 5 above)
Comment (JUNIOR140889) [reply src/modules/payment/screens/payment-screen.tsx]: Done

**Pull request merged**